src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -1983,23 +1982,22 @@ void Llvm::buildCall(GenTreeCall* call)
 
     std::vector<Value*> argVec = std::vector<Value*>();
 
-    GenTreePutArgType* lastArg = nullptr;
+    GenTree* lastArg = nullptr;
     for (CallArg& callArg: call->gtArgs.Args())
     {
-        lastArg = callArg.GetNode()->AsPutArgType();
+        lastArg = callArg.GetNode();
 
-        GenTree* argNode     = lastArg->gtGetOp1();
-        Type*    argLlvmType = getLlvmTypeForCorInfoType(lastArg->GetCorInfoType(), lastArg->GetClsHnd());
+        Type*    argLlvmType = getLlvmTypeForCorInfoType(callArg.GetSignatureCorInfoType(), callArg.GetSignatureClassHandle());
         Value*   argValue;
 
-        if (argNode->OperIs(GT_FIELD_LIST))
+        if (lastArg->OperIs(GT_FIELD_LIST))
```

Comment:
Let's keep `argNode`, `lastArg` isn't as descriptive (and not really correct, since it is actually the current arg).